File: tf2-item-scanner/src/com/minowak/scanner/schema/SchemaParser.java
package com.minowak.scanner.schema;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;
import java.util.List;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

import com.minowak.scanner.gui.MainWindow;

public class SchemaParser {
	private File schemaFile;

	public SchemaParser(File file) {
		this.schemaFile = file;
	}

	public TF2Item[] parse() throws IOException {
		List<TF2Item> list = new LinkedList<TF2Item>();

		BufferedReader br = new BufferedReader(new FileReader(schemaFile));

		String line = null;

		StringBuilder sb = new StringBuilder();

		while((line = br.readLine()) != null) {
			sb.append(line);
		}

		JSONParser parser = new JSONParser();
		try {
			Object schema = parser.parse(sb.toString());
			JSONObject result = (JSONObject)((JSONObject)schema).get("result");
			JSONArray itemsArray = (JSONArray) result.get("items");

			for(int i = 0 ; i < itemsArray.size() ; i++) {
				JSONObject item = (JSONObject)itemsArray.get(i);
				long defindex = (long)item.get("defindex");
				String name = (String)item.get("item_name");
				String imgUrl = ((String)item.get("image_url")).replace("\\", "");

				TF2Item it = new TF2Item(name, defindex, ItemQuality.UNIQUE);
				it.setImgUrl(imgUrl);
				list.add(it);
			}
		} catch (Exception e) {
			MainWindow.LOGGER.severe("Error while parsing schema: " + e.getMessage());
		}

		return list.toArray(new TF2Item[list.size()]);
	}
}

File: tf2-item-scanner/tf2-item-scanner/schema/SchemaParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using tf2_item_scanner.engine;
using Newtonsoft.Json.Linq;
using tf2_item_scanner.utils;

namespace tf2_item_scanner.schema
{
    class SchemaParser
    {
        private string _fileName;

        public SchemaParser(string fileName)
        {
            _fileName = fileName;
        }

        public List<TF2Item> Parse()
        {
            List<TF2Item> items = new List<TF2Item>();

            string schema = File.ReadAllText(_fileName);

            JObject o = JObject.Parse(schema);
            JObject result = (JObject)o["result"];
            JArray array = (JArray)result["items"];

            for (int i = 0; i < array.Count; i++)
            {
                JObject item = (JObject)array[i];
                long defIndex = (long)item["defindex"];
                string name = (string)item["item_name"];
                string imgUrl = (string)item["image_url"];
                imgUrl.TrimStart(new char[] { '\\' });

                TF2Item it = new TF2Item(name, defIndex, ItemQuality.Normal);
                it.Image = imgUrl;
                items.Add(it);
            }

            return items;
        }
    }
}
